### PR TITLE
{IotHub} Update device sdk

### PR DESCRIPTION
**Pull request by anusapan**
## Checklist

- [ ] I have read the [_Submitting Changes_](../blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/master/CONTRIBUTING.md)
- [ ] The title of the PR is clear and informative
- [ ] The appropriate `ChangeLog.md` file(s) has been updated:
    - For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    - A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header -- no new version header should be added
- [ ] The PR does not introduce [breaking changes](../blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [ ] If applicable, the changes made in the PR have proper test coverage
- [ ] For public API changes to cmdlets:
    - [ ] a cmdlet design review was approved for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (_Microsoft internal only_)
    - [ ] the markdown help files have been regenerated using the commands listed [here](../blob/master/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)

### CODE REVIEW

File: src/IotHub/IotHub/IotHub.csproj [hunk 1]
```diff
@@ -11,10 +11,14 @@
...
-    <PackageReference Include="Microsoft.Azure.Devices" Version="1.18.2" />
-    <PackageReference Include="Microsoft.Azure.Devices.Client" Version="1.21.3" />
+    <PackageReference Include="Microsoft.Azure.Devices" Version="1.22.0" />
+    <PackageReference Include="Microsoft.Azure.Devices.Client" Version="1.27.0" />
     <PackageReference Include="Microsoft.Azure.Management.IotHub" Version="2.10.0-preview" />
     <PackageReference Include="Microsoft.Extensions.DependencyInjection.Abstractions" Version="2.2.0" />
   </ItemGroup>
 
+  <ItemGroup>
+    <PackageReference Update="Newtonsoft.Json" Version="12.0.3" />
```
Comment (isra-fel) [inline src/IotHub/IotHub/IotHub.csproj:21]: Newtonsoft.json has already been  a shared reference

**Comment by isra-fel** [comment]
Planning to update newtonsoft.json in Az.Accounts next week, until then this PR is blocked.

**Comment by isra-fel** [comment]
Hi @anusapan we have updated newtonsoft.json in #12620

**Comment by anusapan** [comment]
> Hi @anusapan we have updated newtonsoft.json in #12620

@isra-fel But still I need to update NewtonSoft explicitly in Iothub project, but it is working fine for test project.
Not sure, Is it related or not? but It seems that "tools/Common.Netcore.Dependencies.Test.targets" has been updated with upgraded NewtonSoft but "tools/Common.Netcore.Dependencies.targets" is still using older NewtonSoft.

**Comment by isra-fel** [comment]
`Common.Netcore.Dependencies.targets` doesn't needs to be updated because in production, newtonsoft.json 12.0.3 will be preloaded when importing az.accounts.

Please solve the issue with docs and assembly conflicts with `Az.DeviceProvisioningServices` (details in the log)

File: src/IotHub/IotHub/IotHub.csproj (hunk 1 above)
Comment (anusapan) [reply src/IotHub/IotHub/IotHub.csproj]: removed

**Comment by anusapan** [comment]
> `Common.Netcore.Dependencies.targets` doesn't needs to be updated because in production, newtonsoft.json 12.0.3 will be preloaded when importing az.accounts.
> 
> Please solve the issue with docs and assembly conflicts with `Az.DeviceProvisioningServices` (details in the log)

@isra-fel it's not getting build successfully if we are removing an explicit dependency

File: src/IotHub/IotHub/IotHub.csproj (hunk 1 above)
Comment (anusapan) [reply src/IotHub/IotHub/IotHub.csproj:21]: Added, as discussed

**Comment by isra-fel** [comment]
Pending on #12586

**Comment by isra-fel** [comment]
This PR should be unblocked after #12982 is merged (need to sync with master).

However the 1.0.0 version of Microsoft.Azure.Management.DeviceProvisioningServices has security issue that will fail the check in CI.
```D:\a\1\s\artifacts\Debug\Az.DeviceProvisioningServices\Microsoft.Azure.Management.DeviceProvisioningServices.dll: error BA2022: 'Microsoft.Azure.Management.DeviceProvisioningServices.dll' was signed exclusively with algorithms that WinTrustVerify has flagged as insecure. Cryptographically weak signatures: [digest algorithm: 'sha1NoSign' + digest encryption algorithm: 'RSA']
```
(see https://dev.azure.com/azure-sdk/public/_build/results?buildId=539240&view=logs&j=b9afb22f-0837-5d02-bc24-6c30aee08bcf&t=4b4c9e2b-b8a9-576f-6503-81d0a6dce7ab)

You'll need to downgrade it to 0.10.0-preview like before.

**Review by isra-fel — CHANGES_REQUESTED** [review]
Generally LGTM.
Please update the changelog of iothub otherwise the module will not be picked up during next release

**Review by isra-fel — APPROVED** [review]
LGTM

**Pull request merged**